test/app/src/tokens.js

```diff
@@ -1,15 +1,25 @@
 import { htmlString } from './util';
 
 function tokensHTML(tokens) {
-  if (!tokens || tokens.length < 2) {
+  let accessToken;
+  let idToken;
+  if (!tokens) {
     return '';
+  } else if (Array.isArray(tokens)) {
+    if (tokens.length < 2) {
+      return '';
+    }
+    accessToken = tokens.filter(token => {
```

Comment:
This all feels unnecessarily complex.  Why do we have 3 ways of calling tokensHTML?

Reply:
i am adding support for object instead of array (token renew causes the token to become undefined, and I do not want an undefined entry in the array). Also we will be switching to object syntax for 3.0. The array syntax will be legacy.

Reply:
Oh, okay, I didn't realize tokensHTML was used outside of the tests.

Reply:
Sorry for any confusion, the `tokensHTML` method is used exclusively within the test app, only to display the tokens as HTML for validation by E2E tests or during manual testing. Dealing with the tokens as an array is a pain (and complicated, as you pointed out), especially when/if they can become undefined (this is what I am avoiding by using object syntax). This complexity (and inflexibility) of the array is one of the reasons we are going to switch to an object return type in 3.0 (also we will be able to return additional information, such as state).